File: backend/app/services/instructions_service.py
import os

_INSTRUCTIONS_PATH = os.path.join(
    os.path.dirname(os.path.dirname(__file__)), "assets", "instructions.md"
)


def get_instructions() -> str:
    # Remove the leading "markdown" line if present (artifact from original file)
    with open(_INSTRUCTIONS_PATH, "r", encoding="utf-8") as f:
        content = f.read()
    if content.startswith("markdown\n"):
        content = content[len("markdown\n"):]
    return content


def save_instructions(content: str) -> None:
    with open(_INSTRUCTIONS_PATH, "w", encoding="utf-8") as f:
        f.write(content)

File: server/src/services/instructionsService.ts
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const INSTRUCTIONS_PATH = process.env.INSTRUCTIONS_PATH ||
  path.join(__dirname, '../../../backend/app/assets/instructions.md');

export function getInstructions(): string {
  let content = fs.readFileSync(INSTRUCTIONS_PATH, 'utf-8');
  if (content.startsWith('markdown\n')) {
    content = content.slice('markdown\n'.length);
  }
  return content;
}

export function saveInstructions(content: string): void {
  fs.writeFileSync(INSTRUCTIONS_PATH, content, 'utf-8');
}
